Merge "julien" into "Main"

JulienCarr2 opened the pull request:
Basic Fullness Form function, should work. Updated library data function, had wrote "checkId" instead of "checkValidId", fixed poor handlebar usage (should be somewhat valid HTML now).

JulienCarr2 committed:
Checkbox Form Data

Just trying to take the form data (assuming that the if the checkbox req portion of the form exists, it's a set to "on").

We can discuss this more later, but wanted to push this change since I've been really busy this past week.

JulienCarr2 committed:
Merge branch 'main' into julien

JulienCarr2 committed:
Working Fullness Form

Had to change a lot of the work I did, very sloppily done, but fixed to properly work.

JulienCarr2 committed:
Update libraries.js

JulienCarr2 commented:
Added the editLibrary data function, was easy to just put in

DanielleDauphinais commented on routes/libraries.js:
```diff
@@ -239,150 +256,180 @@ router.route('/:id/survey')
-    let id;
-    
-    // If the library ID is not valid, render the error page with a status code of 400
-    try {
```
Not sure if all this stuff is still needed

JulienCarr2 replied:
Not sure either, kept it for now. Can be removed if needed, just a failsafe in case they try to enter a page id that's invalid.

DanielleDauphinais commented:
looks pretty good

DanielleDauphinais commented on routes/libraries.js:
```diff
@@ -190,39 +208,38 @@ router.route('/:id')
-
     try {
-      res.render('libraries/fullness', { id: id });
+      res.render("libraries/fullness", { id: id });
```
when you render a page it needs to have a title if it's a handlesbar page

JulienCarr2 replied:
Added

DanielleDauphinais commented on routes/libraries.js:
```diff
@@ -239,150 +256,180 @@ router.route('/:id/survey')
+      }
     } catch (e) {
-      return res.status(400).json({ error: e });
+      return res
```
Why would this be a 500 error?

JulienCarr2 replied:
Didn't see this comment until later, sorry that it's outdated.

Didn't know what else to put for now, the code can be changed. I put anything I didn't know what to do as a 500 because it seemed the most vague to spit back if the form was submitted incorrectly (from an external request not from the form), and I wanted to put something that worked with the error page.

Atm it just puts "Internal Server Error" for poorly done data, and now re-renders on failed logic cases (such as no genres for full library, genres for an empty library).

Hope that gives a little bit of insight.

JulienCarr2 committed:
Fullness Form Complete

Should be relatively complete, with basic client side and server side validation.

The clientside javascript is messy, but it works after a lot of testing. hope this helps. Update library will probably be done tomorrow. (Or today).

JulienCarr2 committed:
Added titles to render

Forgot, sorry

JulienCarr2 committed:
Forgot to actually put the updated code, sorry.

JulienCarr2 committed:
Merge branch 'main' into julien

JulienCarr2 commented:
Fixed merge issues, should work with main now.
